### ISSUE TITLE
Neoformat on save snippet - unable to edit afterwards

### ISSUE THREAD

**Issue by sitilge** [sbdchd/neoformat#108]
I'm using the snippet given on the site to run Neoformat on save.
````
" Run Neoformat on save
augroup fmt
	autocmd!
	autocmd BufWritePre * undojoin | Neoformat
augroup END
````
So I edit the file, write it, and then, as far as I see it, the Neoformat is run automatically. However, it is impossible to edit the file afterwards because the changes are reverted back to the formated file.

Is this a misconfiguration or a possible bug? Running `nvim v0.2.0`.

**Comment by sbdchd** [comment]
Not exactly sure of the issue as I don't use the command myself; however, the problem may be that the command is triggering Neoformat _before_ every save, so you make a change, hit save, and then it formats it.

Hence, you changes aren't maintained.

**Comment by sitilge** [comment]
Do you have any idea how to avoid that?

**Comment by sbdchd** [comment]
I am not sure how to have neoformat format on every save and not overwrite your changes.

I would suggest just calling neoformat manually.

**Comment by docwhat** [comment]
If there is an error in the formatting command, then Neoformat will revert it back to the previous one.  Drove me nuts until I figured out what was going on.

I added a toggle as a temporary measure: docwhat/dotfiles@b55f51c7f2

Now that I format on save with Javascript, I need Neomake to run more often to show me problems.

It'd also be good if Neoformat just aborted the format run and left the buffer alone if the command has a non-zero exit code.

**Comment by docwhat** [comment]
I'm assuming I reading this:

>  However, it is impossible to edit the file afterwards because the changes are reverted back to the formated file.

As "the file reverts to the last-saved-state on save."

**Comment by docwhat** [comment]
In my case I'm using `prettier` and it is returning a non-zero exit code (2 in my tests).  But `set verbose=1` shows that Neoformat thinks the exit-code is 0.

**Comment by sbdchd** [comment]
@docwhat that sounds like a bug.

Can you provide a prettier version and a test file?

You might want to make a separate issue as I think this current issue, that @sitilge is having, is due to the `undjoin` before calling `:Neoformat`.

**Comment by docwhat** [comment]
ooh... I think I get it... it's cascading.  If the first fails (`prettier` on my list) it then tries `prettydiff` which mucks everything up.  Huh.  I was expecting it to find the first _existing_ formatter and use that.  

Not to go to the next one if the first returned a non-zero exit code. 

Is that a bug or by design, @sbdchd ?

**Comment by sbdchd** [comment]
Yeah, `Neoformat` tries every formatter defined until one succeeds.

You may want to configure the enabled formatters manually:

```viml
let g:neoformat_enabled_javascript = ['prettier']
```
